uavcan/equipment/gnss/1063.Fix2.uavcan

```diff
@@ -0,0 +1,86 @@
+#
+# GNSS ECEF and LLA navigation solution with uncertainty.
+#
+
+uavcan.Timestamp timestamp         # Global network-synchronized time, if available, otherwise zero
+
+#
+# Time solution.
+# Time standard (GPS, UTC, TAI, etc) is defined in the field below.
+#
+uavcan.Timestamp gnss_timestamp    # GNSS timestamp, if available, otherwise zero
+
+#
+# Time standard used in the GNSS timestamp field.
+#
+uint3 GNSS_TIME_STANDARD_NONE = 0  # Time is unknown
+uint3 GNSS_TIME_STANDARD_TAI  = 1
+uint3 GNSS_TIME_STANDARD_UTC  = 2
+uint3 GNSS_TIME_STANDARD_GPS  = 3
+uint3 gnss_time_standard
+
+void13   # Reserved space
+
+#
+# If known, the number of leap seconds allows to perform conversions between some time standards.
+#
+uint8 NUM_LEAP_SECONDS_UNKNOWN = 0
+uint8 num_leap_seconds
+
+#
+# Position and velocity solution
+#
+int37 longitude_deg_1e8            # Longitude degrees multiplied by 1e8 (approx. 1 mm per LSB)
+int37 latitude_deg_1e8             # Latitude degrees multiplied by 1e8 (approx. 1 mm per LSB on equator)
+int27 height_ellipsoid_mm          # Height above ellipsoid in millimeters
+int27 height_msl_mm                # Height above mean sea level in millimeters
+
+float16[3] ned_velocity            # NED frame (north-east-down) in meters per second
+
+#
+# Fix status
+#
+uint6 sats_used
+
+uint2 STATUS_NO_FIX    = 0
+uint2 STATUS_TIME_ONLY = 1
+uint2 STATUS_2D_FIX    = 2
+uint2 STATUS_3D_FIX    = 3
+uint2 status
+
+#
+# GNSS Mode
+# 
+uint3 MODE_SINGLE      = 0
+uint3 MODE_DGPS        = 1
+uint3 MODE_RTK         = 2
+uint3 MODE_PPP         = 3
+uint3 mode
+
+#
+# GNSS Sub mode
+#
+
+uint5 SUB_MODE_DGPS_OTHER    = 0
+uint5 SUB_MODE_DGPS_SBAS     = 1
+
+uint5 SUB_MODE_RTK_FLOAT     = 0
+uint5 SUB_MODE_RTK_FIXED     = 1
+
+uint5 sub_mode
+
+#
+# Precision
+#
+float16 pdop
+
+float16[<=21] covariance    # Position and velocity covariance. The upper-right
+                            # triangle of a 6x6 symmetric matrix, represented in
+                            # row-major order. Units are m^2 for position, (m/s)^2
+                            # for velocity and m^2/s for position/velocity.
```

Comment:
> Won't an array with 36 float elements use 36 * 2 bytes, and an array with 21 float elements use 21 * 2 bytes bandwidth?

Yes. However, the protocol specifies that special matrix forms (namely scalar, diagonal, and symmetric) can be compressed in order to save bandwidth. So, when a node wants to transmit a diagonal matrix, it can do so in two ways:

* transmit the full matrix, using N<sup>2</sup> elements;
* transmit only the upper-right triangle, using only ((1+N)*N)/2 elements.

The receiving nodes will know N from the data type definition, and therefore they will be able to deduce how to handle the received matrix. Namely:

* if an array of N<sup>2</sup> elements is received, no unpacking is necessary;
* if an array of ((1+N)*N)/2 elements is received, it is mirrored about the diagonal;
* an array of N elements is assigned to the diagonal of a zero NxN matrix;
* an array of 1 element is handled like [1, 1, 1] for the case above;
* an empty array is transformed into an empty matrix.

Libuavcan does provide several convenience methods that automate packing/unpacking: https://github.com/UAVCAN/libuavcan/blob/master/libuavcan/include/uavcan/marshal/array.hpp#L954-L1084. In other libraries the developer will have to pack/unpack manually.